File: trunk/esper/src/main/java/net/esper/view/ViewProcessingException.java
package net.esper.view;

/**
 * This exception is thrown to indicate a problem with a view expression. 
 */
public final class ViewProcessingException extends Exception
{
    /**
     * Constructor.
     * @param message is the error message
     */
    public ViewProcessingException(final String message)
    {
        super(message);
    }

    /**
     * Constructor for an inner exception and message.
     * @param message is the error message
     * @param cause is the inner exception
     */
    public ViewProcessingException(final String message, final Throwable cause)
    {
        super(message, cause);
    }

    /**
     * Constructor.
     * @param cause is the inner exception
     */
    public ViewProcessingException(final Throwable cause)
    {
        super(cause);
    }
}

File: NEsper/view/ViewProcessingException.cs
using System;

namespace net.esper.view
{
    /// <summary>
    /// This exception is thrown to indicate a problem with a view expression.
    /// </summary>

    [Serializable]
    public sealed class ViewProcessingException : Exception
    {
        /// <summary> Constructor.</summary>
        /// <param name="message">is the error message
        /// </param>

        public ViewProcessingException(String message)
            : base(message)
        {
        }

        /// <summary> Constructor for an inner exception and message.</summary>
        /// <param name="message">is the error message
        /// </param>
        /// <param name="cause">is the inner exception
        /// </param>

        public ViewProcessingException(String message, Exception cause)
            : base(message, cause)
        {
        }

        /// <summary> Constructor.</summary>
        /// <param name="cause">is the inner exception
        /// </param>

        public ViewProcessingException(Exception cause)
            : base("", cause)
        {
        }
    }
}